include/deal.II/differentiation/sd/symengine_scalar_operations.h

```diff
@@ -146,8 +146,317 @@ namespace Differentiation
        */
       bool
       is_valid_substitution_symbol(const SymEngine::Basic &entry);
+
+      /**
+       * A convenience function to set the @p value associated with
+       * the @p symbol in the @p substitution_map.
+       *
+       * Using this function ensures that the @p symbol is one that is
+       * valid specifically for the purpose of symbolic substitution.
+       * It must therefore represent a symbol or symbolic derivative,
+       * otherwise an error will be thrown.
+       */
+      void
+      set_value_in_symbol_map(
+        types::substitution_map &                     substitution_map,
+        const SymEngine::RCP<const SymEngine::Basic> &symbol,
+        const SymEngine::RCP<const SymEngine::Basic> &value);
     } // namespace internal
 
+    /**
+     * Return a symbolic map that has the entry key given by @p symbol.
+     * It is expected that all entries be valid symbols or symbolic expressions.
```

Comment:
entries of what? the map? that will only have a single entry. or in the input argument?